### ISSUE TITLE
rows.Err must be checked using rowserrcheck

### ISSUE THREAD

**Issue by yabetsu93** [golangci/golangci-lint#945]
Thank you for creating the issue!

- [ ] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [ ] Yes, I've included all information below (version, config, etc).

Please include the following information:

I upgrade my golangci-lint to 1.23.1 to check unclosed sql connection
for example i have this query:
```rows, err := s.pgdb.Query(SQL, filtersArgs...)
      if err != nil {
          // must passed the linter check
      }
```
but with this current version it keeps on display the error message
```rows.Err must be checked (rowserrcheck)``` despite of existing condition
on the code can someone validate this?
i found this one https://github.com/golang/go/issues/24329 but it doesnt satisfy 
me hoping to hear your reply

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
# paste output here
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
# paste output here
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
# paste output here
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint run -v
# paste output here
```

</details>

**Comment by okhowang** [comment]
I found a style for pass rowserrcheck
```golang
rows, err := db.Query("some sql")
if err != nil {
// error handling
}
defer func() {
    _ = rows.Close()
    _ = rows.Err() // or modify return value
}
for rows.Next() {
  // normal handling
}
```

**Comment by kashifkhan0771** [comment]
> I found a style for pass rowserrcheck
> 
> ```go
> rows, err := db.Query("some sql")
> if err != nil {
> // error handling
> }
> defer func() {
>     _ = rows.Close()
>     _ = rows.Err() // or modify return value
> }
> for rows.Next() {
>   // normal handling
> }
> ```
defer func() {
    _ = rows.Close()
    _ = rows.Err() // or modify return value
}()

should be like this

**Comment by panter-dsd** [comment]
Guys, you fixed the linter error but didn't figure out the real problem in your code.
In this code

```golang
rows, err := db.Query(....)
if err != nil {
  return nil, err
}

for rows.Next() {
  // scan row
}

return result, nil
```
you have a big problem - rows.Next() returns false either when you've reached the end of data or you've gotten an read error. The second case you do not check and this leads you to the fact that only part of the data is returned.
You have to change the code to check Err

```golang
rows, err := db.Query(....)
if err != nil {
  return nil, err
}

for rows.Next() {
  // scan row
}

if err := rows.Err(); err != nil {
  return nil, err
}

return result, nil
```
